lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -416,3 +447,100 @@ def get_num_raft_peers(self) -> int:
         """Return the number of raft peers."""
         raft_config = self._client.sys.read_raft_config()
         return len(raft_config["data"]["config"]["servers"])
+
+    def _get_autounseal_policy_name(self, relation_id: int) -> str:
+        """Return the policy name for the given relation id."""
+        return f"charm-autounseal-{relation_id}"
+
+    def _get_autounseal_approle_name(self, relation_id: int) -> str:
+        """Return the approle name for the given relation id."""
+        return f"charm-autounseal-{relation_id}"
+
+    def _configure_autounseal_policy(self, mount: str, relation_id: int, key_name: str):
+        """Create/update a policy within vault to use the transit key."""
+        mount_policy = textwrap.dedent(f"""
+        path "{mount}/encrypt/{key_name}" {{
+            capabilities = ["update"]
+        }}
+
+        path "{mount}/decrypt/{key_name}" {{
+            capabilities = ["update"]
+        }}
+        """)
+        policy_name = self._get_autounseal_policy_name(relation_id)
+        self._client.sys.create_or_update_policy(
+            policy_name,
+            mount_policy,
+        )
+        return policy_name
+
+    def _get_autounseal_key_name(self, relation_id: int) -> str:
+        """Return the key name for the given relation id."""
+        return str(relation_id)
+
+    def _create_autounseal_key(self, mount_point, relation_id) -> str:
```

Comment:
We're missing variable typing

Reply:
https://github.com/canonical/vault-k8s-operator/pull/365/commits/e930a2b82f4c32b824a45b1387644aac5b6724ea